src/reference/asciidoc/redis.adoc

```diff
@@ -791,12 +791,104 @@ You should set the expiry at a large enough value to prevent this condition, but
+
+Spring Integration 5.4 introduced Reactive Redis Stream outbound channel adapter to write Message payload into Redis stream.
+Outbound Channel adapter uses `ReactiveStreamOperations.add(...)` to send `Message` to stream.
+The following example shows how to use Java Configuration and Service class for Redis Stream Outbound Channel Adapter.
+
+====
+[source,java]
+----
+@Service
+public class MyService {
+
+    @Autowired
+    private FluxMessageChannel messageChannel;
+
+    public void sendMessage(Flux<Object> myFlux) {
+        messageChannel.subscribeTo(myFlux.map(myObj -> new GenericMessage<>(myObj)));
+    }
+}
+
+@Configuration
+@EnableIntegration
+public class MyConfiguration {
+
+    @Bean
+    FluxMessageChannel messageChannel() {
+        return new FluxMessageChannel();
+    }
+
+    @Bean
+    @ServiceActivator(inputChannel = "messageChannel")
+    public ReactiveRedisStreamMessageHandler reactiveValidatorMessageHandler(ReactiveRedisConnectionFactory reactiveRedisConnectionFactory) {
+        return new ReactiveRedisStreamMessageHandler(reactiveRedisConnectionFactory, "myStreamKey");
+    }
+}
+----
+====
+
 [[redis-stream-inbound]]
 === Redis Stream Inbound Channel Adapter
 
-TBD
+Spring Integration 5.4 introduced the Reactive Stream inbound channel adapter for reading messages from a Redis Stream.
+Inbound channel adapter uses `StreamReceiver.receive(...)` or `StreamReceiver.receiveAutoAck` based on
+auto acknowledgement flag to read `Message` from redis stream.
+The following example shows how to use JavaConfiguration for Redis Stream Inbound Channel Adapter.
 
-[[redis-stream-outbound]]
-=== Redis Stream Outbound Channel Adapter
-
-TBD
+====
+[source,java]
+----
+@Configuration
+@EnableIntegration
+public class MyConfiguration {
+
+    @Bean
+    FluxMessageChannel fluxMessageChannel() {
+        return new FluxMessageChannel();
+    }
+
+    @Bean
+    public IntegrationFlow myFlow() {
+        return IntegrationFlows
+                    .from(fluxMessageChannel())
+                    .get();
+    }
+
+    @Bean
+    ReactiveRedisStreamMessageProducer reactiveRedisStreamProducer(ReactiveRedisConnectionFactory reactiveRedisConnectionFactory) {
+    	ReactiveRedisStreamMessageProducer messageProducer =
+                new ReactiveRedisStreamMessageProducer(reactiveRedisConnectionFactory, "myStreamKey");
+        messageProducer.setStreamReceiverOptions( <1>
+                StreamReceiver.StreamReceiverOptions.builder()
+                        .pollTimeout("")
+                        .build());
+        messageProducer.setAutoStartup(""); <2>
+        messageProducer.setAutoAck(""); <3>
+        messageProducer.setCreateConsumerGroup(""); <4>
+        messageProducer.setConsumerGroup(""); <5>
+        messageProducer.setConsumerName(""); <6>
+        messageProducer.setOutputChannel(fluxMessageChannel()); <7>
+        messageProducer.setReadOffset(""); <8>
+        messageProducer.extractPayload(""); <9>
```

Comment:
Hm. I believe it was a copy/paste artefact, but it is wrong direction.
Please, revise values for the options: many of them are `boolean`, but you insert empty string.